Bluepill-cli/Bluepill-cli/Simulator/SimulatorMonitor.m

```diff
@@ -111,9 +112,12 @@ - (void)onTestCasePassedWithName:(NSString *)testName inClass:(NSString *)testCl
 }
 
 - (void)onTestCaseFailedWithName:(NSString *)testName inClass:(NSString *)testClass
-                          inFile:(NSString *)filePath onLineNumber:(NSUInteger)lineNumber wasException:(BOOL)wasException {
-    NSDate *currentTime = [NSDate date];
+                          inFile:(NSString *)filePath onLineNumber:(NSUInteger)lineNumber wasException:(BOOL)wasException saveScreenshot:(BOOL)saveScreenshot {
+    if (saveScreenshot) {
```

Comment:
SimulatorMonitor already has a reference to the `BPConfiguration`, why not just use that instead of creating a new API?
```
if (config.saveScreenshot) {
  saveScreenshot();
}
```